template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx

```diff
@@ -1,174 +1,96 @@
-  const [privyClient, setPrivyClient] = useState<WalletClient | undefined>();
-  const [smartAccount, setSmartAccount] = useState<SmartAccountClient | undefined>();
-  const [ownedTokens, setOwnedTokens] = useState<OwnedTokensType>({});
-  const [showSponsoredModal, setShowSponsoredModal] = useState<boolean>(false);
-  const [transactionHash, setTransactionHash] = useState<string>('');
-  const [mintedItem, setMintedItem] = useState<number>(0);
-
-  // Fetch the NFTs
-  useEffect(() => {
-    const fetchOwnedNFTs = async () => {
-      if (!smartAccount) return;
-      if (!client) return;
-
-      const tokens = await fetchNFTs(smartAccount, client);
-      if (!tokens) return;
-      const tokenMap = createNFTMap(tokens);
-
-      setOwnedTokens(tokenMap);
-    };
-
-    if (authenticated) {
-      void fetchOwnedNFTs();
+  const account = useAccount();
+  const { data: id, writeContracts } = useWriteContracts();
+
+  const handleMint = async () => {
+    console.log('Default url: ', defaultUrl);
+    console.log('account: ', account);
+    if (account.address === undefined) {
+      alert('You need to be signed in to mint an NFT.');
+      console.error('You need to be signed in to mint an NFT.');
+      return;
     }
-  }, [smartAccount, client, authenticated]);
-
-  // Fetch the active wallet
-  useEffect(() => {
-    const embeddedWallet = wallets.find((wallet) => wallet.walletClientType === 'privy');
-    setActiveWallet(embeddedWallet);
-  }, [wallets]);
-
-  // Fetch the privy client
-  useEffect(() => {
-    const fetchPrivyClient = async () => {
-      if (!activeWallet) return;
-
-      try {
-        const eip1193provider = await activeWallet.getEthereumProvider();
-
-        const walletClient = createWalletClient({
-          account: activeWallet.address as `0x${string}`,
-          chain: sepolia, // Replace this with the chain used by your application
-          transport: custom(eip1193provider),
-        });
-
-        setPrivyClient(walletClient);
-      } catch (error) {
-        console.error('Error initializing privyClient:', error);
-      }
-    };
 
-    void fetchPrivyClient();
-  }, [activeWallet]);
-
-  // Create the smart account
-  useEffect(() => {
-    const createSmartAccount = async () => {
-      if (!privyClient) return;
-
-      //@ts-expect-error Privy client makes Account optional whereas walletClientToSmartAccountSigner expects an Account causing type mismatch
-      const signer = walletClientToSmartAccountSigner(privyClient);
-
-      const publicClient = createPublicClient({
-        chain: sepolia, // Replace this with the chain of your app
-        transport: http(rpcUrl),
-      });
-
-      setPublicClient(publicClient);
-
-      const simpleSmartAccountClient = await signerToSimpleSmartAccount(publicClient, {
-        entryPoint: entryPoint,
-        signer: signer,
-        factoryAddress: factoryAddress,
+    try {
+      writeContracts({
```

Comment:
Similar to other contracts, you should generate an Hook for each of them https://github.com/coinbase/build-onchain-apps/blob/main/template/web/app/buy-me-coffee/_contracts/useBuyMeACoffeeContract.ts and avoid to pass the address like this.